### ISSUE TITLE
[HybridApp][Fire Clean Up] Allow C+/Agencies to build Hybrid App

### ISSUE THREAD

**Issue by grgia** [Expensify/App#54321]
See https://expensify.enterprise.slack.com/canvas/C085GD1ET8A

# Context
 > Very few internal engineers had a working local HybridApp build due to multiple issues. More people with access to build means we will have more people available to help next time.

# Problem 
Internal engineers and SWM are the only ones with access to the Mobile-Expensify repository, we should open this up to all C+ and all agencies in order to expand the number of contributors who can help us resolve fires.

<details><summary>Issue Owner</summary>Current Issue Owner: @AndrewGable</details>

**Comment by AndrewGable** [comment]
@cead22 - We want to add all external vendors and the C+ team as read only members to Mobile-Expensify. Thank you!

**Comment by cead22** [comment]
C+ already has read only access to Mobile-Expensify. 

For external vendors, can you share all the GH handles for everyone from SWM and CS, so I can make teams for them, add them, and give them permissions? We usually manage access via teams cause it makes it easier

We have a `Callstack-Automated-Testing` but I don't know if we're using that for something else or if we wanna re-use it
<img width="1280" alt="image" src="https://github.com/user-attachments/assets/ed6c9bd4-b70d-44a6-b9a7-4d27ceb91552" />

**Comment by AndrewGable** [comment]
All the contractors GH names are here: https://github.com/Expensify/Web-Expensify/blob/62d3d0fec560f1b2066525e77629472c23d95d75/_devportal/lib/KernelScheduling/Data.php#L169-L264

Do we have these members on a team already in GitHub?

**Comment by cead22** [comment]
No. `mrousavy` and `perunt` aren't part of the org, and `hannojg` is outside collaborator with direct access to repos, but not part of a team of the Expensify org

**Comment by AndrewGable** [comment]
Going to chat with @mountiny on how we should solve this.

**Comment by mountiny** [comment]
Yeah we haven't really had a good need to keep them as part of the team/organization.

I would say the main reason to not do it is that bee members were joining quite often so it would be extra work for ring0 as we cannot add them without admin rights (could that be changed for this case/team? I assume no)

But they are not really adding new people that often anymore so i would argue we can add them all to the organisation and some generic "expert contributor" team and get it going.

I can look through the web-e list and remove any old devs tomorrow.

Any concerns?

**Comment by cead22** [comment]
The only one is that we pay for these seats. I kinda feel we should close this, cause I don't know if I agree with the problem, but was willing to entertain this if it was going to take a couple of minutes to do



> we should open this up to all C+ and all agencies in order to expand the number of contributors who can help us resolve fires.

Who else are we confident could've and would've helped here?

**Comment by mountiny** [comment]
But if we will add these people to github teams, we wont need the php version anymore. We only keep the contributor names in php arrays to whitelist for the upwork check and we can substitute that to check for the github team membership as source of truth instead of keeping them both. 

I will create a PR to clear out anyone not working on expensify anymore

**Comment by mountiny** [comment]
Just noting that, otherwise I agree with the process 😄

**Comment by AndrewGable** [comment]
I like the idea of creating the source of truth via GitHub teams and using that, I think we have a lot of old cruft in PHP and wont in the GH teams.

**Comment by puneetlath** [comment]
Adding the `Fire-Cleanup` label here. From #fireroom-2024-12-17-hybridapp-crash

**Comment by mountiny** [comment]
I have made a PR to clean up the list and it should now match with who is actively working with us https://github.com/Expensify/Web-Expensify/pull/44913

**Comment by AndrewGable** [comment]
C+ is invited, @cead22 - Do you mind updating the GitHub teams to match Vit's latest update above?

**Comment by cead22** [comment]
- I created the team `external-expert-contributors` and made `callstack` a sub team of that
- I checked all the current and invited members of the `callstack` team and they're all in the whitelist. 
	- 5 of them still need to accept
	- I removed Nathan from the `callstack` team

**Comment by cead22** [comment]
- I created the `software-mansion` team under `external-expert-contributors` and added all the people that were already part of the org to the team
- I created the `margelo` team under `external-expert-contributors` and haven't added anyone because nobody is part of the org
- I gave read only access to the `external-expert-contributors` team to the Mobile-Expensify repo
- Next: get more gh seats and invite the rest

**Comment by blimpich** [comment]
Do we need to ask the folks at Margelo to add themselves to the Github team? https://github.com/orgs/Expensify/teams/margelo

**Comment by AndrewGable** [comment]
I think we need to add them, just waiting on more seats.

**Comment by cead22** [comment]
Yup, I expect to be done with this by EOW, but let me know if this should be more urgent than that

**Comment by mountiny** [comment]
@cead22 @AndrewGable Once we get the seats, lets also please add the rest of the SWM team (for example `adamgrzybowski` or `WojtekBoman` would apprecaite the access to test the navigation changes they work on in hybrid) 

Thank you!

**Comment by mountiny** [comment]
@cead22 Could you please add `pasyukevich` from Callstack to the team as well when you get a minute? Seems like they were missed from the Callstack group

**Comment by mountiny** [comment]
ArekChr - is not working on Expensify anymore so I removed them from the Upwork whitelist (no need to add them back later)

**Comment by AndrewGable** [comment]
Will finish when @cead22 is back from OOO! 😄

**Comment by flodnv** [comment]
The requested seats are now available on GitHub, we have 23 spaces free.

**Comment by AndrewGable** [comment]
Thank you @flodnv !

**Comment by mallenexpensify** [comment]
Is there a discussion specifically for C+ gaining access to Hybrid App? I want to make sure the C+ process doc (and likely onboarding template) is updated.  Also.. curious about how much work this will open up for them (as contributors and C+)

**Comment by AndrewGable** [comment]
C+ has access today via their GitHub team, I don't think there is a plan to open it up to all contributors as that would open source Mobile-Expensify which we have decided not to do.

**Comment by cead22** [comment]
- I invited the rest of the callstack team. There's 21 in total and 7 need to accept the invite
- Same for SWM. There's 21 in total, and 6 need to accept the invite
- For Margelo I invited 9 users and they all need to accept

**Comment by AndrewGable** [comment]
Sweet - I think we are all done here then?

**Comment by AndrewGable** [comment]
Calling it done, reopen if you disagree!

**Comment by cead22** [comment]
Yeah, I was just waiting for GH to provision a few more seats for us before closing, but I just checked and it looks like that was done
